### ISSUE TITLE
How can I use declarative configuration to configure custom logger

### ISSUE THREAD

**Issue by manoojm** [hazelcast/hazelcast-nodejs-client#365]
I saw the code samples and it has an example of to configure the custom logger (winston) in the programmatic configuration approach. Can you please let me know if its possible to do this in a declarative manner i.e. via hazelcast-client.json? 

I checked out the source for LoggingService.initialize and it seems I can provide either an object with log implementation (i.e. ILogger) or 'off', 'default'. 

Basically I would like to capture all logs and redirect them to my application logger.

PS: I understand that documentation work is in-progress but unfortunately my current project timeline is kind of urgent. So, please provide the needful info as soon as possible. Thanks in advance.

**Comment by mustafaiman** [comment]
Hello @manoojm 
Unfortunately specifying logger in declarative configuration is missing. I opened an issue for that(https://github.com/hazelcast/hazelcast-nodejs-client/issues/367). It is a simple fix and I expect it to be available with the next release.

Then, I thought you could extend ConfigBuilder and add your custom logger adapter to generated config object. There I realized `ConfigBuilder` is not public. I opened an issue for that too(https://github.com/hazelcast/hazelcast-nodejs-client/issues/366). This is even a simpler fix and should be available with the next release.

Still, I managed to workaround the problem. It uses an ugly `require` call but I think you can work with this until the real fix arrives.

```javascript
const Client = require('hazelcast-client').Client;

//this is the ugly part
const ConfigBuilder = require('hazelcast-client/lib/config/ConfigBuilder').ConfigBuilder;
/////
const winston = require('winston');

var winstonAdapter = {
    logger: new (winston.Logger)({
        transports: [
            new (winston.transports.Console)()
        ]
    }),

    levels: [
        'error',
        'warn',
        'info',
        'debug',
        'silly'
    ],

    log: function (level, className, message, furtherInfo) {
        this.logger.log(this.levels[level], className + ' ' + message);
    }
};
const configBuilder = new ConfigBuilder();
configBuilder.loadConfig().then(() => {
    const config = configBuilder.build();
    config.properties['hazelcast.logging'] = winstonAdapter;
    return config;
}).then((config) => {
    Client.newHazelcastClient(config);
});
```

**Comment by manoojm** [comment]
Thanks for the quick reply. I can confirm that this works. Looking forward to the fix in the next release.
Having a config builder as public would be very powerful.
